### PR TITLE
Replace Prettier with ESLint

### PR DESCRIPTION
**Pull request by haykam821**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,11 +21,42 @@
...
+    "@typescript-eslint/eslint-plugin": "^4.28.1",
+    "@typescript-eslint/parser": "^4.28.1",
     "@vercel/ncc": "^0.28.6",
+    "eslint": "^7.29.0",
+    "eslint-config-haykam": "^1.10.0",
+    "eslint-plugin-extra-rules": "^0.0.0-development",
+    "eslint-plugin-html": "^6.1.2",
+    "eslint-plugin-jsdoc": "^32.3.4",
+    "eslint-plugin-json": "^2.1.2",
+    "eslint-plugin-markdown": "^2.2.0",
```
Comment (modmuss50) [inline package.json:33]: why do we need html, jsdoc, json and markdown?

File: package.json [hunk 2]
```diff
@@ -21,11 +21,42 @@
   "devDependencies": {
     "@octokit/types": "^6.17.2",
     "@types/jest": "^26.0.23",
+    "@typescript-eslint/eslint-plugin": "^4.28.1",
+    "@typescript-eslint/parser": "^4.28.1",
     "@vercel/ncc": "^0.28.6",
+    "eslint": "^7.29.0",
+    "eslint-config-haykam": "^1.10.0",
```
Comment (modmuss50) [inline package.json:28]: What does this do? I would prefer to stick to defalts

File: src/main.ts [hunk 3]
```diff
@@ -3,23 +3,23 @@ import { getOctokit } from "@actions/github";
...
-        github.rest,
-        parseInt(core.getInput("issue-number", { required: true }))
-      );
-      break;
-    default:
-      throw new Error("Unknown context: " + context);
-  }
+	switch (context) {
+		case "yarn-update-base":
+			await yarnUpdateBase(
```
Comment (modmuss50) [inline src/main.ts:15]: ah, good spot, this await is needed 👍

**Review by modmuss50 — COMMENTED** [review]
Formatting is better yes, just a lot of packages that im not sure we need for a small project like this. The simpliler the better as this is something that I dont want to have to battle to maintain like every other npm/js project ive touched.

The .vscode file will also need updating, I can do that if needed.

File: package.json (hunk 1 above)
Comment (haykam821) [reply package.json:33]: Unfortunately, due to the limitations of ESLint, these dependencies are forced upon any project using a configuration that depends on it.

File: package.json (hunk 2 above)
Comment (haykam821) [reply package.json:28]: That's my ESLint configuration which enables [lots of opinionated rules](https://github.com/haykam821/ESLint-Config-Haykam/blob/dc44224bc9a7339f70472873f197b77733501614/eslint-config.js). It's generally in line with the existing Java conventions such as using tabs and double quotes.

File: src/main.ts (hunk 3 above)
Comment (haykam821) [reply src/main.ts:15]: That was just the [`require-await`](https://eslint.org/docs/rules/require-await) rule, which wouldn't be caught by the recommended configuration.

File: package.json (hunk 2 above)
Comment (modmuss50) [reply package.json:28]: My main worry is keeping this maintained and updated, what we have now is super simple with no custom config, so nothing to worry about.

This isnt a java project so using java's conventions makes little sense to me? surely using javascripts would be a better way to go? The main point of a linter is just to ensure that contributions meet a standard to save a reviewer pointing out styling issues.

What are eslint's defaults like? Why is it better than what I setup?

**Comment by modmuss50** [comment]
i dont think we need this.

**Pull request closed without merging**